openedx/core/djangoapps/content_libraries/api/libraries.py

```diff
@@ -692,3 +689,42 @@ def revert_changes(library_key: LibraryLocatorV2, user_id: int | None = None) ->
 
     # Call the event handlers as needed.
     tasks.wait_for_post_revert_events(draft_change_log, library_key)
+
+
+def get_backup_task_status(
+    library_key: LibraryLocatorV2,
+    user_id: int,
+    task_id: str | None = None
+) -> dict | None:
+    """
+    Get the status of a library backup task.
+
+    Returns a dictionary with the following keys:
+        - state: One of "Pending", "Exporting", "Succeeded", "Failed"
+        - url: If state is "Succeeded", the URL where the exported .zip file can be downloaded. Otherwise, None.
+    If no task is found, returns None.
+    """
+    task_status = None
+    if task_id:
+        # Get the specific task by ID
+        try:
+            task_status = UserTaskStatus.objects.get(task_id=task_id, user_id=user_id)
+        except UserTaskStatus.DoesNotExist:
+            return None
+    else:
+        # Get the latest task for this user and library
+        task_status = UserTaskStatus.objects.filter(
+            user_id=user_id,
+            name__contains=str(library_key)
```

Comment:
So at a low level, I think we'd want to filter by `task_class` (so there's no mixing of backup and restore tasks), and put a bound on `created` (automated processes may make many, many tasks over time).

But at a higher level, is this part of the API here to service the use case where someone wanders away from the backup page and then comes back to it? Because if so, I'd rather we just not handle that case yet, and require the user to stay on the page. It shouldn't take that long to create the backup, and it's not clear to me how long a backup should remain "good", or what exactly the logic for invalidating a backup download should be, if we allow people to access historical backups.